### ISSUE TITLE
PBS qsub fails when the job name starts with a dot 

### ISSUE THREAD

**Issue by drejom** [nextflow-io/nextflow#228]
pbs_version = PBSPro_11.2.0.113417

Submitting a job script as a hidden file fails:

`qsub .command.run`
```
qsub: illegal -N value
usage: qsub [-a date_time] [-A account_string] [-c interval
       [-C directive_prefix] [-e path] [-h ] [-I [-X]] [-j oe|eo] [-J X-Y[:Z]]
       [-k o|e|oe] [-l resource_list] [-m mail_options] [-M user_list]
       [-N jobname] [-o path] [-p priority] [-q queue] [-r y|n]
       [-S path] [-u user_list] [-W otherattributes=value...]
       [-v variable_list] [-V ] [-z] [script | -- command [arg1 ...]]
      qsub --version
```

Specifying the job name outside the script or removing the leading '.' on the script name works
```
cp .command.run command.run
qsub command.run
```
or
`qsub -N name .command.run`

**Comment by pditommaso** [comment]
I've uploaded a patch that should solve this problem. You may want to try it defining the following variable in your environment `NXF_VER=0.22.2-SNAPSHOT`, then running nextflow usual.

**Comment by drejom** [comment]
Success! Thanks a lot for fixing this so quickly Paolo
Cheers
Den

**Comment by pditommaso** [comment]
Great. I will prepare a new release soon.

**Comment by pditommaso** [comment]
Released with version `0.22.2`

**Comment by drejom** [comment]
Hi Paolo,
Might have jumped the gun here. I've moved to a newer cluster with `pbs_version = PBSPro_12.2.4.142262` and despite the new release get the  "qsub: illegal -N value" error when the job name is longer than 15 characters: 

```
qsub -N "nf-star_(A14_R1_val_1.fq.gz_A14_R2_val_2.fq.gz)" .command.run
qsub: illegal -N value
```
but 
```
qsub -N "nf-star_(A14_R1" .command.run
723931.mgmt1
```
Works fine. 

Seems this is the case for older(?) PBSPro versions. From the man page
```
-N     name Declares a name for the job.  The name specified may
                  be up to and including 15 characters in length.  It
                  must consist of printable characters with the first
                  character alphabetic.

                  If the -N option is not specified, the job name will
                  be the base name of the job script file specified on
                  the command line.  If no script file name was
                  specified and the script was read from the standard
                  input, then the job name will be set to STDIN.
```

**Comment by pditommaso** [comment]
Oh, any plan to switch to another cluster manager? :) 

I will patch this. In the meanwhile as a workaround you can add the following rule in the `nextflow.config` file. 

    executor.jobName = { task.name.size()>15 ? task.name.substring(0,15) : task.name }

**Comment by drejom** [comment]
No joy yet, this admits whitespace to the job name, so it still fails with the error
```
qsub -N 'trimG (A13)' .command.run
qsub: directive error: -N trimG (A13) 
```

**Comment by pditommaso** [comment]
try this 

    executor.jobName = { task.name.size()>15 ? task.name.substring(0,15).replace(' ','') : task.name }

**Comment by drejom** [comment]
Same error I’m afraid - hasn’t replaced the whitespace

```
 qsub -N 'trimG (A2)' .command.run
 qsub: directive error: -N trimG (A2) 
```

**Comment by pditommaso** [comment]
Because it's shorter then 15 chars .. try this 

```
executor.jobName = { def x=task.name.replace(' ',''); x.size()>15 ? x.substring(0,15) : x } 
```

**Comment by drejom** [comment]
Of course, that makes sense…and works! Thanks again Paolo!
Cheers
Den

**Comment by pditommaso** [comment]
I've uploaded a new snapshot fixing this problem. You may want to give a try adding the following var in your env, then launching NF as usual. 

    NXF_VER=0.22.3-SNAPSHOT

**Comment by pditommaso** [comment]
Included with version `0.22.3`

**Comment by drejom** [comment]
Sorry, only just got a chance to test it - all good! Thanks again!
